src/subtype.c

```diff
@@ -2740,6 +2760,97 @@ static jl_value_t *intersect_varargs(jl_vararg_t *vmx, jl_vararg_t *vmy, jl_sten
     return ii;
 }
 
+static int normalize_vb_offset(jl_svec_t **params JL_REQUIRE_ROOTED_SLOT, jl_tvar_t *v, jl_varbinding_t *vb, jl_vararg_t *ii, jl_stenv_t *e)
+{
+    assert(!jl_has_typevar(ii->T, v));
+    if (ii->N != (jl_value_t*)v) {
+        return 1;
+    }
+    ssize_t pos = jl_svec_len(*params) - 1;
+    if (vb->offset < 0) {
+        // Insert additional copies of the type to make up the
+        // `offset` difference.
+        *params = jl_svec_copy_resize(*params, jl_svec_len(*params) - vb->offset);
+        for (size_t i = 0; i < -vb->offset; ++i) {
+            jl_svecset(*params, pos + i, ii->T);
+        }
+        jl_svecset(*params, pos + -vb->offset, ii);
+    } else {
+        assert(vb->offset > 0);
+        // Go backwards over the parameter list and verify that
+        // they are all equal to the vararg type.
+        for (ssize_t i = pos - 1; i > pos - 1 - vb->offset; i--) {
+            if (!forall_exists_equal(jl_svecref(*params, i), ii->T, e)) {
+                return 0;
+            }
+        }
+        jl_svecset(*params, pos - vb->offset, ii);
+        ((jl_svec_t*)*params)->length -= vb->offset;
```

Comment:
This can move an object from big to pooled, which may make the GC abort. Instead make a copy.